components/provider/src/structs/likelysubtags.rs

```diff
@@ -0,0 +1,15 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/master/LICENSE ).
+use icu_locid::LanguageIdentifier;
+use serde::{Deserialize, Serialize};
+
+pub mod key {
+    use crate::resource::ResourceKey;
+    pub const LIKELY_SUBTAGS_V1: ResourceKey = resource_key!(likelysubtags, "likelysubtags", 1);
+}
+
+#[derive(Debug, PartialEq, Clone, Default, Deserialize, Serialize)]
+pub struct LikelySubtagsV1 {
+    pub entries: Vec<(LanguageIdentifier, LanguageIdentifier)>,
```

Comment:
Just to verify: is this data model, a vector of langid pairs, the most correct and efficient data model for likely subtags?  Maybe it is, and that's great.  I just want to make sure that you chose this data model because it is a good data model, not because it happens to be the same as you get from CLDR JSON, because you do *not need to* mimic CLDR JSON.

Reply:
This is roughly what Zibi did in https://github.com/zbraniecki/unic-locale/tree/master/unic-langid-impl/src/likelysubtags, so I think it is a reasonable starting place. His implementation breaks things up into multiple smaller tables which seems like it might improve search performance. This might also be a good place for a trie or maybe something like perfect hashing.

I was planning to look at performance after this landed and follow up with a separate pull request if I can make improvements. If you think it makes more sense, I can look at the performance aspects now.

Reply:
OK, this is fine as a baseline.  It makes it easier to get an objective measurement on benchmarks (all types of benchmarks) when an optimization is proposed.